plugins/inputs/logfile/globpath/globpath.go

```diff
@@ -33,7 +33,7 @@ func Compile(path string) (*GlobPath, error) {
 
 	// if there are no glob meta characters in the path, don't bother compiling
 	// a glob object or finding the root directory. (see short-circuit in Match)
-	if !out.hasMeta || !out.hasSuperMeta {
+	if !out.hasMeta && !out.hasSuperMeta {
```

Comment:
nit: I believe his should be an OR. We only need to compile the glob in the `hasSuperMeta` case. With an AND, we would be compiling the glob and not using it.

Reply:
Also if you look at the diff for https://github.com/aws/amazon-cloudwatch-agent/pull/815/commits/d81957b82fef78ccd2686e2b542110b3afb4c599, I think this also illustrates it. If I flip it to an OR condition here, then the valid glob `/nested1/nested2/{nested, foo}.txt` does not return the expected result. It matches zero elements when it should match `nested1/nested2/nested.txt`

Reply:
You're right. I was thinking we were implementing the `hasMeta` differently. In a different implementation, we used
```go
// isMeta determines whether the rune is a glob character.
func isMeta(char rune) bool {
	return syntax.Special(byte(char))
}
```
from "github.com/gobwas/glob/syntax"